File: github-mvi-android/app/src/release/java/com/rightpoint/mvi/example/injection/BuildTypeModule.java
package com.rightpoint.mvi.example.injection;

import android.os.StrictMode;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import dagger.Module;
import dagger.Provides;
import timber.log.Timber;

@Module
abstract class BuildTypeModule {
    @Provides
    static StrictMode.ThreadPolicy providesThreadPolicy() {
        return StrictMode.ThreadPolicy.LAX;
    }

    @Provides
    static StrictMode.VmPolicy providesVmPolicy() {
        return StrictMode.VmPolicy.LAX;
    }

    @Provides
    static Timber.Tree providesTree() {
        return new Timber.Tree() {
            @Override
            protected void log(int priority,
                               @Nullable String tag,
                               @NotNull String message,
                               @Nullable Throwable t) {
                // Do nothing
            }
        };
    }
}

File: {{ cookiecutter.repo_name }}/app/src/release/java/{{ cookiecutter.package_dir }}/BuildTypeModule.kt
package {{ cookiecutter.package_name }}

import android.os.StrictMode

import dagger.Module
import dagger.Provides
import timber.log.Timber

@Module
internal object BuildTypeModule {
    @Provides
    @JvmStatic
    fun providesThreadPolicy(): StrictMode.ThreadPolicy {
        return StrictMode.ThreadPolicy.LAX
    }

    @Provides
    @JvmStatic
    fun providesVmPolicy(): StrictMode.VmPolicy {
        return StrictMode.VmPolicy.LAX
    }

    @Provides
    @JvmStatic
    fun providesTree(): Timber.Tree {
        return object : Timber.Tree() {
            override fun log(priority: Int, tag: String?, message: String, t: Throwable?) {
                // Do nothing
            }
        }
    }
}
